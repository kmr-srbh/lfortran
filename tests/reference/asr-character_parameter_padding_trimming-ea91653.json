{
    "basename": "asr-character_parameter_padding_trimming-ea91653",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/character_parameter_padding_trimming.f90",
    "infile_hash": "3ac17fa68feb60423eaa66ebbb88a3e18da1a72d0c7e8ed8ac8de022",
    "outfile": null,
    "outfile_hash": null,
    "stdout": null,
    "stdout_hash": null,
    "stderr": "asr-character_parameter_padding_trimming-ea91653.stderr",
    "stderr_hash": "7435da5a3769b1b237c9999f77f6e8766604928d96cda18974248e80",
    "returncode": 2
}