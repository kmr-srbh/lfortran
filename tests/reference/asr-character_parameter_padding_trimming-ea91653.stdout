(TranslationUnit
    (SymbolTable
        1
        {
            character_parameter_padding_trimming:
                (Program
                    (SymbolTable
                        2
                        {
                            p_pad:
                                (Variable
                                    2
                                    p_pad
                                    []
                                    Local
                                    (ArrayConstant
                                        12
                                        ["2   ", "2   ", "2   "]
                                        (Array
                                            (Character 1 4 () PointerString)
                                            [((IntegerConstant 1 (Integer 4) Decimal)
                                            (IntegerConstant 3 (Integer 4) Decimal))]
                                            FixedSizeArray
                                        )
                                        ColMajor
                                    )
                                    (ArrayConstant
                                        12
                                        ["2   ", "2   ", "2   "]
                                        (Array
                                            (Character 1 4 () PointerString)
                                            [((IntegerConstant 1 (Integer 4) Decimal)
                                            (IntegerConstant 3 (Integer 4) Decimal))]
                                            FixedSizeArray
                                        )
                                        ColMajor
                                    )
                                    Parameter
                                    (Array
                                        (Character 1 4 () PointerString)
                                        [((IntegerConstant 1 (Integer 4) Decimal)
                                        (IntegerConstant 3 (Integer 4) Decimal))]
                                        FixedSizeArray
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            p_trim:
                                (Variable
                                    2
                                    p_trim
                                    []
                                    Local
                                    (ArrayConstant
                                        3
                                        ["2", "2", "2"]
                                        (Array
                                            (Character 1 1 () PointerString)
                                            [((IntegerConstant 1 (Integer 4) Decimal)
                                            (IntegerConstant 3 (Integer 4) Decimal))]
                                            FixedSizeArray
                                        )
                                        ColMajor
                                    )
                                    (ArrayConstant
                                        3
                                        ["2", "2", "2"]
                                        (Array
                                            (Character 1 1 () PointerString)
                                            [((IntegerConstant 1 (Integer 4) Decimal)
                                            (IntegerConstant 3 (Integer 4) Decimal))]
                                            FixedSizeArray
                                        )
                                        ColMajor
                                    )
                                    Parameter
                                    (Array
                                        (Character 1 1 () PointerString)
                                        [((IntegerConstant 1 (Integer 4) Decimal)
                                        (IntegerConstant 3 (Integer 4) Decimal))]
                                        FixedSizeArray
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            x_pad:
                                (Variable
                                    2
                                    x_pad
                                    []
                                    Local
                                    (StringConstant
                                        "apple"
                                        (Character 1 5 () PointerString)
                                    )
                                    (StringConstant
                                        "apple   "
                                        (Character 1 8 () PointerString)
                                    )
                                    Parameter
                                    (Character 1 8 () PointerString)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            x_trim:
                                (Variable
                                    2
                                    x_trim
                                    []
                                    Local
                                    (StringConstant
                                        "apple"
                                        (Character 1 5 () PointerString)
                                    )
                                    (StringConstant
                                        "ap"
                                        (Character 1 2 () PointerString)
                                    )
                                    Parameter
                                    (Character 1 2 () PointerString)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            y_pad:
                                (Variable
                                    2
                                    y_pad
                                    []
                                    Local
                                    (StringConstant
                                        "Ball"
                                        (Character 1 4 () PointerString)
                                    )
                                    (StringConstant
                                        "Ball      "
                                        (Character 1 10 () PointerString)
                                    )
                                    Parameter
                                    (Character 1 10 () PointerString)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            y_trim:
                                (Variable
                                    2
                                    y_trim
                                    []
                                    Local
                                    (StringConstant
                                        "Ball"
                                        (Character 1 4 () PointerString)
                                    )
                                    (StringConstant
                                        "Bal"
                                        (Character 1 3 () PointerString)
                                    )
                                    Parameter
                                    (Character 1 3 () PointerString)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            z_pad:
                                (Variable
                                    2
                                    z_pad
                                    [x_pad
                                    y_pad]
                                    Local
                                    (StringConcat
                                        (StringConcat
                                            (Var 2 x_pad)
                                            (Var 2 y_pad)
                                            (Character 1 18 () PointerString)
                                            (StringConstant
                                                "apple   Ball      "
                                                (Character 1 18 () PointerString)
                                            )
                                        )
                                        (Var 2 x_pad)
                                        (Character 1 26 () PointerString)
                                        (StringConstant
                                            "apple   Ball      apple   "
                                            (Character 1 26 () PointerString)
                                        )
                                    )
                                    (StringConstant
                                        "apple   Ball      apple       "
                                        (Character 1 30 () PointerString)
                                    )
                                    Parameter
                                    (Character 1 30 () PointerString)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            z_trim:
                                (Variable
                                    2
                                    z_trim
                                    [x_trim
                                    y_trim]
                                    Local
                                    (StringConcat
                                        (StringConcat
                                            (Var 2 x_trim)
                                            (Var 2 y_trim)
                                            (Character 1 5 () PointerString)
                                            (StringConstant
                                                "apBal"
                                                (Character 1 5 () PointerString)
                                            )
                                        )
                                        (Var 2 x_trim)
                                        (Character 1 7 () PointerString)
                                        (StringConstant
                                            "apBalap"
                                            (Character 1 7 () PointerString)
                                        )
                                    )
                                    (StringConstant
                                        "apBal"
                                        (Character 1 5 () PointerString)
                                    )
                                    Parameter
                                    (Character 1 5 () PointerString)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    character_parameter_padding_trimming
                    []
                    [(Print
                        [(StringConcat
                            (StringConcat
                                (StringConstant
                                    "|"
                                    (Character 1 1 () PointerString)
                                )
                                (Var 2 x_pad)
                                (Character 1 9 () PointerString)
                                (StringConstant
                                    "|apple   "
                                    (Character 1 9 () PointerString)
                                )
                            )
                            (StringConstant
                                "|"
                                (Character 1 1 () PointerString)
                            )
                            (Character 1 10 () PointerString)
                            (StringConstant
                                "|apple   |"
                                (Character 1 10 () PointerString)
                            )
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(Var 2 y_pad)]
                        ()
                        ()
                    )
                    (Print
                        [(StringFormat
                            ()
                            [(Var 2 p_pad)]
                            FormatFortran
                            (Character -1 0 () PointerString)
                            ()
                        )]
                        ()
                        ()
                    )
                    (Print
                        [(Var 2 x_trim)]
                        ()
                        ()
                    )
                    (Print
                        [(Var 2 y_trim)]
                        ()
                        ()
                    )
                    (Print
                        [(Var 2 z_trim)]
                        ()
                        ()
                    )
                    (Print
                        [(StringFormat
                            ()
                            [(Var 2 p_trim)]
                            FormatFortran
                            (Character -1 0 () PointerString)
                            ()
                        )]
                        ()
                        ()
                    )]
                )
        })
    []
)
